File: src/main/java/no/nav/syfo/domain/Gjennomfoering.java
package no.nav.syfo.domain;

import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(fluent = true)
public class Gjennomfoering {
    public String gjennomfoeringStatus;
    public Boolean paaAnnetSted;
    public Boolean medMerTid;
    public Boolean medHjelp;
    public String kanBeskrivelse;
    public String kanIkkeBeskrivelse;

    public enum KanGjennomfoeres {
        KAN,
        KAN_IKKE,
        TILRETTELEGGING
    }
}

File: src/main/kotlin/no/nav/syfo/api/v3/domain/oppfolgingsplan/Gjennomfoering.kt
package no.nav.syfo.api.v3.domain.oppfolgingsplan

data class Gjennomfoering (
    val kanGjennomfoeres: String,
    val paaAnnetSted: Boolean? = null,
    val medMerTid: Boolean? = null,
    val medHjelp: Boolean? = null,
    val kanBeskrivelse: String? = null,
    val kanIkkeBeskrivelse: String? = null,
)
